Microsoft.Azure.CosmosRepository.CleanArchitecture/EtagMappedRepository.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CosmosRepository.Options;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.CosmosRepository.CleanArchitecture;
+
+public class EtagMappedRepository<TItem, TMapped>
+    : IEtagMappedRepository<TItem, TMapped>
+    where TItem : IItemWithEtag
+    where TMapped : class
+{
+    private readonly IRepository<TItem> _itemRepository;
+    private readonly IItemMapper<TItem, TMapped> _itemMapper;
+    private readonly IOptionsMonitor<RepositoryOptions> _repositoryOptions;
+    private readonly IMappedItemMapper<TMapped, TItem> _mappedItemMapper;
+    private string? _etag;
+    private IDisposable _optionsMonitor;
+
+    public string? Etag => _etag;
```

Comment:
How would this handle collections?

Reply:
I have updated it to track multiple etags in one using block.